src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx

```diff
@@ -230,6 +235,42 @@ export class DashboardAppController {
       }
     };
 
+    const showFilterBar = () =>
+      $scope.model.filters.length > 0 || !dashboardStateManager.getFullScreenMode();
+
+    const getNavBarProps = () => {
+      const isFullScreenMode = dashboardStateManager.getFullScreenMode();
+      return {
+        appName: 'dashboard',
+        config: isFullScreenMode ? undefined : $scope.topNavMenu,
+        noPadding: isFullScreenMode,
+        showSearchBar: !isFullScreenMode,
+        showFilterBar: showFilterBar() && !isFullScreenMode,
```

Comment:
This hides the filter bar in fullscreen mode - currently the filter bar is shown there. `isVisible` is not the same as fullscreen mode, it's about whether the chrome is shown or not (which also happens when in embedded mode)